x/swap/keeper/deposit.go

```diff
@@ -0,0 +1,62 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/kava-labs/kava/x/swap/types"
+)
+
+func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, coinA sdk.Coin, coinB sdk.Coin) error {
+	poolName := types.PoolName(coinA.Denom, coinB.Denom)
+
+	_, found := k.GetPool(ctx, poolName)
+	if found {
+		return sdkerrors.Wrap(types.ErrNotImplemented, fmt.Sprintf("can not deposit into existing pool '%s'", poolName))
+	}
+
+	err := k.depositAllowed(ctx, poolName, coinA, coinB)
+	if err != nil {
+		return err
+	}
+
+	amount := sdk.NewCoins(coinA, coinB)
+	err = k.supplyKeeper.SendCoinsFromAccountToModule(ctx, depositor, types.ModuleAccountName, amount)
+	if err != nil {
+		return err
+	}
+
+	k.initializePool(ctx, depositor, coinA, coinB)
```

Comment:
error value not checked.  I would also put this before the call the `SendCoinsFromAccountToModule`

Reply:
resolved in other work